src/services/configuration/storedConfigurationService.ts

```diff
@@ -0,0 +1,320 @@
+   */
+  public async selectProject(name: string): Promise<void> {
+    const configuration = await this.getOrCreateConfiguration();
+    const project = configuration.projects.find((project) => project.name === name);
+    if (!project) {
+      this.logger.warn(`Couldn't find project ${name}.`);
+      return;
+    }
+    if (project.isActive) {
+      this.logger.log(`Project ${name} is already active.`);
+      return;
+    }
+
+    await this.updateProject({ name, isActive: true });
+  }
+
+  /**
+   * Deselects a project to target with future commands.
+   *
+   * @param {string} name The project to deselect.
+   */
+  public async deselectProject(name: string): Promise<void> {
+    const configuration = await this.storageService.read(FileConstants.configFileName);
+    if (configuration) {
+      const project = configuration.projects.find((project) => project.name === name);
+      if (!project) {
+        this.logger.warn(`Couldn't find project ${name}.`);
+        return;
+      }
+      if (!project.isActive) {
+        this.logger.log(`Project ${name} is already inactive.`);
+        return;
+      }
+
+      await this.updateProject({ name, isActive: false });
+    }
+  }
+
+  /**
+   * Updates the properties on a project.
+   *
+   * @param {OnlyRequire<ProjectConfiguration, "name">} options The options to update the project with.
+   * Besides name, all arguments are optional.
+   */
+  public async updateProject(options: OnlyRequire<ProjectConfiguration, "name">): Promise<void> {
+    const configuration = await this.getOrCreateConfiguration();
+    const matches: ProjectConfiguration[] = [];
+    const others: ProjectConfiguration[] = [];
+    configuration.projects.forEach((project) =>
+      project.name === options.name ? matches.push(project) : others.push(project),
+    );
+
+    if (matches.length === 1) {
+      const match = matches[0];
+      const mergedConfiguration: ProjectConfiguration = { ...match, ...options };
+
+      others.push(mergedConfiguration);
+
+      configuration.projects = others;
+
+      await this.storageService.write(FileConstants.configFileName, configuration);
+      return;
+    } else if (matches.length > 1) {
+      this.logger.warn(
+        `Something's wrong with your config file:\nTried to update project ${options}\nBut multiple projects ${matches
+          .map((match) => match.name)
+          .join()} found: ${matches}`,
+      );
+    } else {
+      this.logger.warn(`No project named ${options.name} found.`);
+    }
+  }
+
+  //#endregion
+
+  /**
+   * Gets the current configuration, or creates a new one if none is found.
+   *
+   * @returns {Configuration} A configuration.
+   */
+  private async getOrCreateConfiguration(): Promise<Configuration> {
+    return (
+      (await this.storageService.read(FileConstants.configFileName)) ?? {
+        playbooks: [],
+        projects: [],
+        appInsights: {
+          instrumentationKey: "6e4b4afa-05f9-4a1f-a5f8-a95958b53f60",
```

Comment:
any reason this is hard-coded?

Reply:
Yeah, this is hard-coded in our defaults today as well. Couldn't figure out how to get the app to send telemetry without distributing the key in some shape or form :\